File: Assets/LuaModules/Graphics/GraphicsModule.lua
--require LuaCanvas
GraphicsModule = {}
local canvasModule = require 'Canvas'
local canvas = nil
function GraphicsModule.new()
    canvas = canvasModule.new(9999, 5)
    return GraphicsModule
end

function GraphicsModule:SpawnElement(name)
    canvas:SpawnElement(canvas, name)
end

function GraphicsModule:SpawnButton(name, positionx, positiony, width, height, onclick)
    canvas:SpawnButton(name, positionx, positiony, width, height, onclick)
    GraphicsModuleProxy.SpawnButton(name, positionx, positiony, width, height, onclick)
end

function GraphicsModule:GetElementByName(name)
    return canvas:GetElementByName(name)
end

function GraphicsModule:MoveElement(object, newpositionX, newpositionY)
    canvas:MoveElement(object, newpositionX, newpositionY)
    GraphicsModuleProxy.MoveElement(object.name, newpositionX, newpositionY)
end

function GraphicsModule:Update()
    canvas:Update()
end


return GraphicsModule

File: Assets/Modules/Graphics/GraphicsModule.cs
using System;
using Luncay.Core;
using MoonSharp.Interpreter;
using UnityEngine;

namespace Modules.Graphics
{
    public class GraphicsModule
    {
        private readonly CanvasManager manager;

        public GraphicsModule(CanvasManager manager)
        {
            this.manager = manager;
        }

        public void Spawnbutton(string name, Vector2 position, float width, float height, Action onclick)
        {
            manager.SpawnButton(name, position, width, height, onclick);
        }

        public void MoveElement(string name, Vector2 newPosition)
        {
            manager.MoveElement(name, newPosition);
        }

        public void StartCoroutine()
        {
            FakeCoreModule.fakeCoreModule.StartTheMovement(); 
        }

        public void Update()
        {
            CanvasManager.instance.DebugLog();
        }

        public void MoveButtonWithDoTween(string name, float endPositionX, float endPositionY, float time, DynValue callback)
        {
            CanvasManager.instance.MoveWithDotween(name, endPositionX, endPositionY, time, callback);
        }
    }
}